geth/common/services/whisper.go

```diff
@@ -0,0 +1,80 @@
+package services
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/whisper/whisperv5"
+)
+
+// Whisper interface for Whisper service.
+type Whisper interface {
```

Comment:
Am in agreement with @adambabik  here, why are we creating interfaces for all these structs:

- Is it for concurrent-safe usage?

I fear we really will be polluting the interfaces unnecessarily here.

Reply:
How can we do proper unit tests for complicated structs without the mocks?